test/feedcontainer.cpp

```diff
@@ -249,14 +251,25 @@ TEST_CASE("Correctly sorts feeds", "[FeedContainer]")
 		REQUIRE(sorted_feeds[3]->get_order() == 10);
 		REQUIRE(sorted_feeds[4]->get_order() == 33);
 	}
+}
+
+TEST_CASE("sort_feeds() sorts by feed's first tag if `feed-sort-order` "
+	"is \"fisttag\"",
+	"[FeedContainer]")
+{
+	FeedContainer feedcontainer;
+	ConfigContainer cfg;
+	Cache rsscache(":memory:", &cfg);
+	const auto feeds = get_five_empty_feeds(&rsscache);
+	feedcontainer.set_feeds(feeds);
 
 	feeds[0]->set_tags({"tag1"});
 	feeds[1]->set_tags({"Taggy"});
 	feeds[2]->set_tags({"tag10"});
 	feeds[3]->set_tags({"taggy"});
 	feeds[4]->set_tags({"tag2"});
 
-	SECTION("by firsttag asc") {
+	SECTION("`firsttag-asc` -- tags in dictionary order") {
```

Comment:
Is there a definition for `dictionary order`?

It's apparently not lexicographical order as that would have resulted in `"tag10"` to be between `"tag1"` and `"tag2"`.
(I guess there is some special handling of numbers in the tag name but I didn't look into the sorting implementation, yet).

Reply:
I think "dictionary order" is a colloquial name for lexicographical order; I used it because it sounds less scary. Wikipedia [backs me up on this](https://en.wikipedia.org/wiki/Lexicographical_order).

The order where "tag10" is placed after "tag2" is called "natural" ([Wikipedia again](https://en.wikipedia.org/wiki/Natural_sort_order)). We use it for article titles (PR landed a few years ago), but didn't add it anywhere else (I don't have an excuse for that).

So I think this description is correct.

With that said, if you know how to succinctly describe the order used here, I can use that and avoid the problem of definitions altogether. The best I came up with is "tags sorted alphabetically, no special handling for numbers", but that's rather long and IMHO a bit confusing in its own way.

Reply:
> The order where "tag10" is placed after "tag2" is called "natural" (Wikipedia again).

I didn't know, thanks!

> We use it for article titles (PR landed a few years ago), but didn't add it anywhere else

It looks like it is [used for `firsttag`-sorting](https://github.com/newsboat/newsboat/blob/5d0de6ccfc179f7c85bb8f5c663c3f4ab4569fd1/src/feedcontainer.cpp#L24-L46) as well (using `utils::strnaturalcmp()`).

I think the current descriptions look fine so no need to change anything.
Thanks for the helpful wikipedia references.

Reply:
Wait, if `firsttag` uses natural order, then the test description is wrong. Looks like I got so wrapped up explaining myself that I lost track of what you're saying: you pointed out that "tag10" is after "tag2", but I proceeded to explain about natural order and all that stuff.

Sorry about that! I'll change the description to say "natural order".

Reply:
I also see I got directions wrong here… "T" comes before "t" in ASCII, so normal order is "Taggy then tag1", whereas "tag1 then Taggy" is reverse.

Reply:
Good catch. I've left a comment for another location where I think the direction is specified inocrrectly.
Other than that, looks good.